tests/cxx/enum_int_type.cc

```diff
@@ -0,0 +1,20 @@
+//===--- enum_int_type.cc - test input file for iwyu ----------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cstdint>
+
+enum class Test : int32_t {
+    VALUE1,
+    VALUE2
+};
+/**** IWYU_SUMMARY
+
+(tests/cxx/enum_int_type.cc has correct #includes/fwd-decls)
```

Comment:
Please see https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUStyle.md#regression-tests, and particularly https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUStyle.md#force-diagnostics.

I'd even go so far as suggest you shouldn't use `<cstdint>` here, because we're not in control of its contents -- it could (and probably does) include `stdint.h`, which in turn could include `bits/inttypes.h` or something.

So I'd make this three files:
```
// enum_base-i1.h
typedef int int_t;

// enum_base-d1.h
#include "tests/cxx/enum_base-i1.h"

// enum_base.cc
#include "tests/cxx/enum_base-d1.h"

...
```

It's significantly more complicated, but it gives a diagnostic on the use of `int_t` in `enum_base.cc`, which you can use to verify that it's finding that specific use.